### PR TITLE
MissionController: Add takeoff only with supported vehicles

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
This code is incorrect.  The Vehicle::takeoffVehicleSupported is for guided commands not for mission commands. What problem are you trying to fix here?

**Comment by patrickelectric** [comment]
Thank you for reply @DonLakeFlyer, this patch remove takeoff in mission commands in vehicles that don't have this feature, like ArduSub.
Can you say the best approach to solve this problem ?

**Comment by DonLakeFlyer** [comment]
ArduSub should implement FirmwarePlugin::supportedMissionCommands. Then something like that:

```
    if (_visualItems->count() == 1) {
        MAV_COMMAND takeoffCmd = _controllerVehicle->vtol() ? MAV_CMD_NAV_VTOL_TAKEOFF : MAV_CMD_NAV_TAKEOFF;
        if (_controllerVehicle->firwarePlugin()->supportedMissionCommands().contains(takeoffCmd)) {
            newItem->setCommand((MavlinkQmlSingleton::Qml_MAV_CMD)takeoffCmd;
    }
```

**Comment by patrickelectric** [comment]
@DonLakeFlyer is this implementation correct ?

File: src/MissionManager/MissionController.cc [hunk 1]
```diff
@@ -1033,8 +1036,9 @@ void MissionController::_recalcWaypointLines(void)
         // If we still haven't found the first coordinate item and we hit a takeoff command, link back to home
         if (firstCoordinateItem &&
                 item->isSimpleItem() &&
-                (qobject_cast<SimpleMissionItem*>(item)->command() == MavlinkQmlSingleton::MAV_CMD_NAV_TAKEOFF ||
-                 qobject_cast<SimpleMissionItem*>(item)->command() == MavlinkQmlSingleton::MAV_CMD_NAV_VTOL_TAKEOFF)) {
+                (!_controllerVehicle->firmwarePlugin()->supportedMissionCommands().contains(MAV_CMD_NAV_TAKEOFF) ||
```
Comment (DonLakeFlyer) [inline src/MissionManager/MissionController.cc]: This part should not change. This is testing whether a takeoff command is already in the mission.

Comment (patrickelectric) [reply src/MissionManager/MissionController.cc]: without this, when planning the mission, the home do not connect with the first item.

Comment (DonLakeFlyer) [reply src/MissionManager/MissionController.cc]: Ok, I get it now. I wasn't understanding the code correctly. This is right.

**Comment by DonLakeFlyer** [comment]
This is all good to go now

**Pull request merged**